docs/metasploit-framework.wiki/Creating-Your-First-PR.md

```diff
@@ -0,0 +1,122 @@
+# Creating Your First PR - An Intro To Git and the PR Process
+## Intro
+Congratulations fellow traveler, so your interested in contributing to Metasploit eh? Well welcome aboard, its going to be a fun ride!
+You'll learn lots along the way but here are some tips and tricks that should help you get started with making your first PR request
+whilst also avoiding some common pitfalls and learning how some of our systems work.
+
+## Initial Steps and Important Notes
+The rest of this guide assumes you have already followed the steps at [Setting Up A Developer Environment](https://r-7.co/MSF-DEV) in order to get
+a fork of Metasploit set up and ready to run, and that you have added in your SSH keys, set up Ruby and optionally the PostgreSQL database, and
+done any custom shortcuts you wish to configure.
+
+## Getting the Latest Version of Metasploit Framework
+Before making any new contributions, you will want to sure you are running the latest version of Metasploit Framework.
+To do this run `git checkout master && git fetch && git pull`. This will check out the `master` branch, then fetch all
```

Comment:
`git fetch` without another argument is going to fetch the origin. IIRC, the preferred setup is to clone your fork in which case to update from Rapid7, you'd have to have had that setup as another remote and fetch it from there. That's typically the `upstream` remote.

If they followed the step [here](https://docs.metasploit.com/docs/development/get-started/setting-up-a-metasploit-development-environment.html#set-up-your-local-copy-of-the-repository) from the docs you're suggesting then I don't think they'll actually get Rapid7's code.

I'd suggest updating that part to explicitly include the remote and maybe mention what that remote is. Something to the effect of:

```
`git checkout master && git fetch upstream && git pull` where "upstream" is the Rapid7 remote. This can be verified using the command `git remote get-url upstream` and should be `git@github.com:rapid7/metasploit-framework.git
```

Reply:
Huh interesting it worked for someone who followed that guide however I do fully agree with your statement here and think the extra detail surrounding this is needed. Thanks for pointing this out, I'll update this now.

Reply:
Added in new explanation in https://github.com/rapid7/metasploit-framework/pull/16949/commits/2bfcac8d8630c70097cd0b029271ec9c25e522d6 using your template, let me know if that works for you.